src/azure-cli/azure/cli/command_modules/cosmosdb/aaz/latest/cosmosdb/postgres/_check_name_availability.py

```diff
@@ -0,0 +1,180 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "cosmosdb postgres check-name-availability",
+)
+class CheckNameAvailability(AAZCommand):
+    """Checks availability of a cluster name. Cluster names should be globally unique; at least 3 characters and at most 40 characters long; they must only contain lowercase letters, numbers, and hyphens; and must not start or end with a hyphen.
```

Comment:
For the property validation, usually we will define it in swagger aaz-dev-tools will add validation automatically. This is an example for your reference: https://github.com/Azure/azure-rest-api-specs/blob/6914c95db58c8dd037faf07bcc21475fc19c57f3/specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json#L1930-L1936

```json
        "name": {
          "type": "string",
          "description": "Template parameter name.",
          "minLength": 1,
          "maxLength": 256,
          "pattern": "^[A-Za-z0-9-._]+$"
        },
```

Reply:
This validation also checks if the name is used or not across all clusters. Per your suggestion, we'd need to update the swagger file first, I can update azure-cli code once we check in the swagger change to azure-rest-api-specs repo.